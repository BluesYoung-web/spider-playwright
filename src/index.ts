/*
 * @Author: zhangyang
 * @Date: 2022-09-24 16:23:50
 * @LastEditTime: 2024-07-11 17:41:20
 * @Description: 
 */
export * from './spiders/holiday';
export * from './spiders/iptv';
export * from './spiders/ccmihb';