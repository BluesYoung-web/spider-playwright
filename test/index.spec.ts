/*
 * @Author: zhangyang
 * @Date: 2022-09-24 15:57:54
 * @LastEditTime: 2022-12-10 11:30:13
 * @Description: 
 */
import { describe, it, expect } from 'vitest';
import { getFullYearData, isHoliday } from '../src';


describe('spider demo', () => {
  it('get holiday data', async () => {
    expect(await getFullYearData('2022')).toMatchSnapshot();
  });

  it('National Day', async () => {
    expect(await isHoliday('20221001')).toBe(true);
    expect(await isHoliday('20221002')).toBe(true);
    expect(await isHoliday('20221003')).toBe(true);
    expect(await isHoliday('20221004')).toBe(true);
    expect(await isHoliday('20221005')).toBe(true);
    expect(await isHoliday('20221006')).toBe(true);
    expect(await isHoliday('20221007')).toBe(true);

    expect(await isHoliday('20221008')).toBe(false);
    expect(await isHoliday('20221009')).toBe(false);
    expect(await isHoliday('20221010')).toBe(false);
    expect(await isHoliday('20221011')).toBe(false);
    expect(await isHoliday('20221012')).toBe(false);
    expect(await isHoliday('20221013')).toBe(false);
    expect(await isHoliday('20221014')).toBe(true);

    expect(await isHoliday('20221015')).toBe(true);
    expect(await isHoliday('20221016')).toBe(true);
    
    expect(await isHoliday('20221028')).toBe(true);

  });

  it('2023', async () => {
    expect(await getFullYearData('2023')).toMatchSnapshot();
  });
});