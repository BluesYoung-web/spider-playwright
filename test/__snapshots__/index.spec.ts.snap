// Vitest Snapshot v1

exports[`spider demo > 2023 1`] = `
{
  "20230101": true,
  "20230102": true,
  "20230106": true,
  "20230107": true,
  "20230108": true,
  "20230113": true,
  "20230114": true,
  "20230115": true,
  "20230120": true,
  "20230121": true,
  "20230122": true,
  "20230123": true,
  "20230124": true,
  "20230125": true,
  "20230126": true,
  "20230127": true,
  "20230203": true,
  "20230204": true,
  "20230205": true,
  "20230210": true,
  "20230211": true,
  "20230212": true,
  "20230217": true,
  "20230218": true,
  "20230219": true,
  "20230224": true,
  "20230225": true,
  "20230226": true,
  "20230303": true,
  "20230304": true,
  "20230305": true,
  "20230310": true,
  "20230311": true,
  "20230312": true,
  "20230317": true,
  "20230318": true,
  "20230319": true,
  "20230324": true,
  "20230325": true,
  "20230326": true,
  "20230331": true,
  "20230401": true,
  "20230402": true,
  "20230405": true,
  "20230407": true,
  "20230408": true,
  "20230409": true,
  "20230414": true,
  "20230415": true,
  "20230416": true,
  "20230421": true,
  "20230422": true,
  "20230428": true,
  "20230429": true,
  "20230430": true,
  "20230501": true,
  "20230502": true,
  "20230503": true,
  "20230505": true,
  "20230507": true,
  "20230512": true,
  "20230513": true,
  "20230514": true,
  "20230519": true,
  "20230520": true,
  "20230521": true,
  "20230526": true,
  "20230527": true,
  "20230528": true,
  "20230602": true,
  "20230603": true,
  "20230604": true,
  "20230609": true,
  "20230610": true,
  "20230611": true,
  "20230616": true,
  "20230617": true,
  "20230618": true,
  "20230622": true,
  "20230623": true,
  "20230624": true,
  "20230630": true,
  "20230701": true,
  "20230702": true,
  "20230707": true,
  "20230708": true,
  "20230709": true,
  "20230714": true,
  "20230715": true,
  "20230716": true,
  "20230721": true,
  "20230722": true,
  "20230723": true,
  "20230728": true,
  "20230729": true,
  "20230730": true,
  "20230804": true,
  "20230805": true,
  "20230806": true,
  "20230811": true,
  "20230812": true,
  "20230813": true,
  "20230818": true,
  "20230819": true,
  "20230820": true,
  "20230825": true,
  "20230826": true,
  "20230827": true,
  "20230901": true,
  "20230902": true,
  "20230903": true,
  "20230908": true,
  "20230909": true,
  "20230910": true,
  "20230915": true,
  "20230916": true,
  "20230917": true,
  "20230922": true,
  "20230923": true,
  "20230924": true,
  "20230929": true,
  "20230930": true,
  "20231001": true,
  "20231002": true,
  "20231003": true,
  "20231004": true,
  "20231005": true,
  "20231006": true,
  "20231013": true,
  "20231014": true,
  "20231015": true,
  "20231020": true,
  "20231021": true,
  "20231022": true,
  "20231027": true,
  "20231028": true,
  "20231029": true,
  "20231103": true,
  "20231104": true,
  "20231105": true,
  "20231110": true,
  "20231111": true,
  "20231112": true,
  "20231117": true,
  "20231118": true,
  "20231119": true,
  "20231124": true,
  "20231125": true,
  "20231126": true,
  "20231201": true,
  "20231202": true,
  "20231203": true,
  "20231208": true,
  "20231209": true,
  "20231210": true,
  "20231215": true,
  "20231216": true,
  "20231217": true,
  "20231222": true,
  "20231223": true,
  "20231224": true,
  "20231229": true,
  "20231230": true,
  "20231231": true,
}
`;

exports[`spider demo > get holiday data 1`] = `
{
  "20220101": true,
  "20220102": true,
  "20220103": true,
  "20220107": true,
  "20220108": true,
  "20220109": true,
  "20220114": true,
  "20220115": true,
  "20220116": true,
  "20220121": true,
  "20220122": true,
  "20220123": true,
  "20220128": true,
  "20220131": true,
  "20220201": true,
  "20220202": true,
  "20220203": true,
  "20220204": true,
  "20220205": true,
  "20220206": true,
  "20220211": true,
  "20220212": true,
  "20220213": true,
  "20220218": true,
  "20220219": true,
  "20220220": true,
  "20220225": true,
  "20220226": true,
  "20220227": true,
  "20220304": true,
  "20220305": true,
  "20220306": true,
  "20220311": true,
  "20220312": true,
  "20220313": true,
  "20220318": true,
  "20220319": true,
  "20220320": true,
  "20220325": true,
  "20220326": true,
  "20220327": true,
  "20220401": true,
  "20220403": true,
  "20220404": true,
  "20220405": true,
  "20220408": true,
  "20220409": true,
  "20220410": true,
  "20220415": true,
  "20220416": true,
  "20220417": true,
  "20220422": true,
  "20220423": true,
  "20220429": true,
  "20220430": true,
  "20220501": true,
  "20220502": true,
  "20220503": true,
  "20220504": true,
  "20220506": true,
  "20220508": true,
  "20220513": true,
  "20220514": true,
  "20220515": true,
  "20220520": true,
  "20220521": true,
  "20220522": true,
  "20220527": true,
  "20220528": true,
  "20220529": true,
  "20220603": true,
  "20220604": true,
  "20220605": true,
  "20220610": true,
  "20220611": true,
  "20220612": true,
  "20220617": true,
  "20220618": true,
  "20220619": true,
  "20220624": true,
  "20220625": true,
  "20220626": true,
  "20220701": true,
  "20220702": true,
  "20220703": true,
  "20220708": true,
  "20220709": true,
  "20220710": true,
  "20220715": true,
  "20220716": true,
  "20220717": true,
  "20220722": true,
  "20220723": true,
  "20220724": true,
  "20220729": true,
  "20220730": true,
  "20220731": true,
  "20220805": true,
  "20220806": true,
  "20220807": true,
  "20220812": true,
  "20220813": true,
  "20220814": true,
  "20220819": true,
  "20220820": true,
  "20220821": true,
  "20220826": true,
  "20220827": true,
  "20220828": true,
  "20220902": true,
  "20220903": true,
  "20220904": true,
  "20220909": true,
  "20220910": true,
  "20220911": true,
  "20220912": true,
  "20220916": true,
  "20220917": true,
  "20220918": true,
  "20220923": true,
  "20220924": true,
  "20220925": true,
  "20220930": true,
  "20221001": true,
  "20221002": true,
  "20221003": true,
  "20221004": true,
  "20221005": true,
  "20221006": true,
  "20221007": true,
  "20221014": true,
  "20221015": true,
  "20221016": true,
  "20221021": true,
  "20221022": true,
  "20221023": true,
  "20221028": true,
  "20221029": true,
  "20221030": true,
  "20221104": true,
  "20221105": true,
  "20221106": true,
  "20221111": true,
  "20221112": true,
  "20221113": true,
  "20221118": true,
  "20221119": true,
  "20221120": true,
  "20221125": true,
  "20221126": true,
  "20221127": true,
  "20221202": true,
  "20221203": true,
  "20221204": true,
  "20221209": true,
  "20221210": true,
  "20221211": true,
  "20221216": true,
  "20221217": true,
  "20221218": true,
  "20221223": true,
  "20221224": true,
  "20221225": true,
  "20221230": true,
  "20221231": true,
}
`;
